From 82cb9e8f70de7d78e9bda9273fc60cb41bb15265 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Mon, 5 Jun 2023 22:56:53 +0200
Subject: [PATCH 0279/1141] arm64: dts: st: add ltdc on stm32mp251

The LCD-TFT Display Controller (LTDC) handles display composition, scaling and
rotation. It provides a parallel digital RGB flow to be used by display
interfaces.

Signed-off-by: Philippe Cornu <philippe.cornu@foss.st.com>
Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Change-Id: I1c3d6287c8652dca7cabbeb336e2c10aa3509f5d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/305993
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Domain-Review: Yannick FERTRE <yannick.fertre@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 3a603ad2ed22..c74be4a74f0c 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -1724,6 +1724,19 @@ trigger@4 {
 				};
 			};
 
+			ltdc: display-controller@48010000 {
+				compatible = "st,stm32-ltdc";
+				reg = <0x48010000 0x400>;
+				st,syscon = <&syscfg>;
+				interrupts = <GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 159 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>;
+				clock-names = "bus", "lcd";
+				resets = <&rcc LTDC_R>;
+				feature-domains = <&rifsc STM32MP25_RIFSC_LTDC_CMN_ID>;
+				status = "disabled";
+			};
+
 			combophy: phy@480c0000 {
 				compatible = "st,stm32mp25-combophy";
 				reg = <0x480c0000 0x1000>;
-- 
2.39.2

